build.gradle

```diff
@@ -214,6 +236,53 @@ task deploy() {}
 
 subprojects {
 
+  if (file('src/main/java').directory) {
+    apply plugin: 'com.jfrog.bintray'
+    apply plugin: 'maven-publish'
+
+    publishing {
+      publications {
+        mavenJava(MavenPublication) {
+          groupId "tech.pegasys.ethsigner.internal"
+          version "${project.version}"
+          from components.java
+          artifact sourcesJar
+          versionMapping {
+            usage('java-api') { fromResolutionOf('runtimeClasspath') }
+            usage('java-runtime') { fromResolutionResult() }
+          }
+          pom {
+            name = "Ethsigner - ${project.name}"
+            url = 'http://github.com/PegaSysEng/ethsigner'
+            licenses {
+              license {
+                name = 'The Apache License, Version 2.0'
+                url = 'http://www.apache.org/licenses/LICENSE-2.0.txt'
+              }
+            }
+            scm {
+              connection = 'scm:git:git://github.com/PegaSysEng/ethsigner.git'
+              developerConnection = 'scm:git:ssh://github.com/PegaSysEng/ethsigner.git'
+              url = 'https://github.com/PegaSysEng/ethsigner'
+            }
+          }
+        }
+      }
+    }
+
+    bintray {
+      user = bintrayUser
+      key = bintrayKey
+
+      publications = ['mavenJava']
+      override = version.endsWith('SNAPSHOT')
+
+      publish = true
+
+      pkg = bintrayPackage
+    }
+  }
+
   tasks.withType(Test) {
     // If GRADLE_MAX_TEST_FORKS is not set, use half the available processors
     maxParallelForks = (System.getenv('GRADLE_MAX_TEST_FORKS') ?: (Runtime.runtime.availableProcessors().intdiv(2) ?: 1)).toInteger()
```

Comment:
In Pantheon the "apply plugin: 'maven-publish'" was removed from the "tasks.withType(JavaCompile)" does it make sense to do the same?

Reply:
Tested and removed!